src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php

```diff
@@ -319,10 +348,26 @@ protected function loadCacheDriver($cacheName, $objectManagerName, array $cacheD
 
                 return $cacheDriverServiceId;
             case 'memcached':
-                $memcachedClass = !empty($cacheDriver['class']) ? $cacheDriver['class'] : '%'.$this->getObjectManagerElementName('cache.memcached.class').'%';
-                $memcachedInstanceClass = !empty($cacheDriver['instance_class']) ? $cacheDriver['instance_class'] : '%'.$this->getObjectManagerElementName('cache.memcached_instance.class').'%';
-                $memcachedHost = !empty($cacheDriver['host']) ? $cacheDriver['host'] : '%'.$this->getObjectManagerElementName('cache.memcached_host').'%';
-                $memcachedPort = !empty($cacheDriver['port']) ? $cacheDriver['port'] : '%'.$this->getObjectManagerElementName('cache.memcached_port').'%';
+                if ($container->hasParameter($this->getObjectManagerElementName('cache.memcached.class'))) {
+                    @trigger_error(
+                        'Usage of class parameter for memcache class is deprecated since version 4.2 and will be removed in 5.0. Use configuration values instead.',
+                        E_USER_DEPRECATED
+                    );
+                }
+                if ($container->hasParameter($this->getObjectManagerElementName('cache.memcached_instance.class'))) {
+                    @trigger_error(
+                        'Usage of class parameter for memcached_instance class is deprecated since version 4.2 and will be removed in 5.0. Use configuration values instead.',
+                        E_USER_DEPRECATED
+                    );
+                }
+                $memcachedClass = !empty($cacheDriver['class']) ? $cacheDriver['class'] : '%' . $this->getObjectManagerElementName('cache.memcached.class') . '%';
```

Comment:
We should not expect the class to come from the config data. Each bundle knows what the classes are. We don't want them to make it configurable in their semantic config (that would be even worse than the DI parameter).